### PR TITLE
Add minia + seqwish + vg assembly workflow

### PR DESCRIPTION
**Pull request by heuermh**
Work in progress.

Adds the `minia` + `seqwish` + `vg` assembly workflow adapted from https://github.com/hpobio-lab/viral-analysis/blob/master/nextflow/assemble.nf.

Still left to do:
- [x] Add new programs to environment and build new Docker image
- [x] Add citations
- [ ] Add new docs
- [ ] Confirm correct files are published to results directory
- [ ] Should the variant calls go through variant annotation and evaluation steps?

Ping @ekg @drpatelh for review.

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/viralrecon branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/viralrecon)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by heuermh** [comment]
Is there a test that is running to fail the `--skip_variants` check?  I am not seeing a failure when run locally, and the CI logs aren't helpful.

**Comment by drpatelh** [comment]
Re-triggered the tests @heuermh. Seen these sort of failures in the past :smirk: Think it may be due to download issues with the test data.

**Comment by drpatelh** [comment]
Citations and tools added in https://github.com/nf-core/viralrecon/pull/62

**Comment by ekg** [comment]
How is this working?

**Comment by drpatelh** [comment]
Ah, because @heuermh added biocontainer paths that will now need to be removed because I added them to the main pipeline container in https://github.com/nf-core/viralrecon/pull/62:
https://github.com/heuermh/viralrecon/blob/f426d46ae559d748395ecdc96f9a03562e2a3066/main.nf#L1939

**Comment by heuermh** [comment]
Sorry for the git-mess . . . force push never causes me troubles, except for when it does

File: main.nf [hunk 1]
```diff
@@ -1929,6 +1934,236 @@ process UNICYCLER_QUAST {
...
+
+    input:
+    set val(sample), file(graph) from ch_minia_call_variants
+    output:
+    set val(sample), file("${sample}.vg"), file("${sample}.xg"), file("${sample}.vcf") into ch_minia_variants
+
+    """
+    vg view -Fv $graph > ${sample}.vg
+    vg convert -x ${sample}.vg > ${sample}.xg
+    vg deconstruct -p NC_045512.2 ${sample}.xg > ${sample}.vcf
```
Comment (heuermh) [inline main.nf:2042]: Notice this hard-coded reference sequence accession `NC_045512.2` here, that will have to come out

**Comment by drpatelh** [comment]
No worries! You may want to wait until I have merged all of those processes though because it may become a mess sorting out those conflicts too. Sorry 😓

File: main.nf [hunk 2]
```diff
@@ -1929,6 +1934,236 @@ process UNICYCLER_QUAST {
...
+    !params.skip_assembly && 'minia' in assemblers
+
+    input:
+    set val(sample), val(single_end), file(fasta) from ch_minia_overlap_scaffolds
+    file reference from ch_fasta_minia_overlap_scaffolds.collect()
+    output:
+    set val(sample), file(fasta), file("${sample}.paf") into ch_minia_induce_graph
+
+    """
+    minimap2 -c -x asm20 $reference $fasta > ${sample}.paf
```
Comment (heuermh) [inline main.nf:1996]: Should we pull any `minimap2` parameters used here out to config?

File: main.nf [hunk 3]
```diff
@@ -1929,6 +1934,236 @@ process UNICYCLER_QUAST {
...
+    set val(sample), val(single_end), file("*scaffolds.fa") into ch_minia_overlap_scaffolds,
+                                                                 ch_minia_quast,
+                                                                 ch_minia_blast,
+                                                                 ch_minia_abacas,
+                                                                 ch_minia_plasmidid
+
+    script:
+    if (single_end) {
+        """
+        minia -kmer-size 51 -abundance-min 20 -in $reads -out ${sample}.k51.a20
```
Comment (heuermh) [inline main.nf:1964]: Should we pull any `minia` parameters used here out to config?  Is there any utility in doing multi-k here?

File: main.nf (hunk 1 above)
Comment (drpatelh) [reply main.nf:2042]: Yep. I have used this bit of code to get the prefix using bash for now:
https://github.com/drpatelh/nf-core-viralrecon/blob/124823d293e8806f5dd558851ab6c4654e21a8b9/main.nf#L1526

However, this wont work for multi-fasta files @ekg?

**Comment by drpatelh** [comment]
Thanks @heuermh Going to merge this now to collate all of the changes in https://github.com/nf-core/viralrecon/pull/63

**Comment by heuermh** [comment]
Thank you, @drpatelh!

**Pull request merged**